packages/connectivity/src/scp-cf/destination/http-proxy-util.ts

```diff
@@ -42,28 +46,30 @@ export function proxyStrategy(destination: HttpDestination): ProxyStrategy {
     return 'private-link';
   }
 
-  const destinationProtocol = getProtocolOrDefault(destination);
-  if (!getProxyEnvValue(destinationProtocol)) {
-    logger.debug(
-      `Could not find proxy settings for ${destinationProtocol} in the environment variables - no proxy used.`
-    );
-    return 'no-proxy';
-  }
+  if (isHttpDestination(destination)) {
+    const destinationProtocol = getProtocolOrDefault(destination);
+    if (!getProxyEnvValue(destinationProtocol)) {
+      logger.debug(
+        `Could not find proxy settings for ${destinationProtocol} in the environment variables - no proxy used.`
+      );
+      return 'no-proxy';
+    }
 
-  if (getNoProxyEnvValue().includes(destination.url)) {
-    logger.debug(
-      `Destination URL ${
-        destination.url
-      } is in no_proxy list: ${getNoProxyEnvValue()} - no proxy used.`
-    );
-    return 'no-proxy';
-  }
+    if (getNoProxyEnvValue().includes(destination.url)) {
+      logger.debug(
+        `Destination URL ${
+          destination.url
+        } is in no_proxy list: ${getNoProxyEnvValue()} - no proxy used.`
+      );
+      return 'no-proxy';
+    }
 
-  if (getProxyEnvValue(destinationProtocol)) {
-    logger.debug(
-      `Proxy settings for ${destinationProtocol} are found in environment variables.`
-    );
-    return 'internet';
+    if (getProxyEnvValue(destinationProtocol)) {
+      logger.debug(
+        `Proxy settings for ${destinationProtocol} are found in environment variables.`
+      );
+      return 'internet';
+    }
```

Comment:
[pp] It seems to me that it would be very helpful to extract this code into a separate function with a speaking name, you have to read too much code to understand the flow of this function in my opinion.